clang/lib/AST/ExprConstant.cpp

```diff
@@ -11536,6 +11536,9 @@ bool ArrayExprEvaluator::VisitCXXParenListOrInitListExpr(
   LValue Subobject = This;
   Subobject.addArray(Info, ExprToVisit, CAT);
   auto Eval = [&](const Expr *Init, unsigned ArrayIndex) {
+    if (Init->isValueDependent())
```

Comment:
@shafik https://github.com/llvm/llvm-project/issues/112140#issuecomment-2412422576 I've looked into it, and it seems the value-dependent Init expression

https://github.com/llvm/llvm-project/blob/6f21a7bdeeca84bcc7cf94878e17b5d7ee7b4083/clang/lib/AST/ExprConstant.cpp#L11620-L11621

is causing the assertion failure. Would it be acceptable to try running an evaluation on a dependent expression?

Reply:
This looks reasonable, but please again I have have stated before in other PRs the summary needs to be more detailed. Explain that we are passing ArrayFillers to EvaluateInPlace when they are dependent .... and then explain what the fix is.

Thank you

Reply:
@shafik Thanks for the feedback. I've updated the PR summary.